### PR TITLE
Add debug operation to clear a message locally

### PR DESCRIPTION
**Pull request by Valodim**
While working on #1386 I needed a way to clear messages easily. This adds a menu item to the messagelist context menu which does just that, and which is only shown if `K9.DEBUG`.

I'm not sure if we merge that sort of thing, but thought I'd put it here anyways. Having those options available makes it more easy to test corner cases, increasing the probability that people test them.

### CODE REVIEW

**Comment by philipwhiuk** [comment]
I think we might add want it more formally once we support (re)downloading a message / message body. Occasionally people encounter scenarios where the message was downloaded in a corrupt fashion.

We already support wiping the entire message store or a single folder.

**Comment by Valodim** [comment]
This only clears the message body, not the message itself, to get to a state where it has to be downloaded for view. I don't think that is a valid use case outside of debugging, .

**Comment by cketti** [comment]
The option needs a name that is more descriptive. Otherwise I like the idea.

In the future I want to get rid of the other developer options ("Clear messages", "Recreate data", "clear pending actions") in release builds, too.

**Comment by Valodim** [comment]
More descriptive name, like what?

**Comment by cketti** [comment]
"Clear message body"?

**Comment by Valodim** [comment]
You got it

File: k9mail/src/main/java/com/fsck/k9/controller/MessagingController.java [hunk 1]
```diff
@@ -3648,6 +3649,27 @@ public void run() {
 
     }
 
+    @SuppressLint("NewApi") // used for debugging only
+    public void debugClearMessagesLocally(final List<LocalMessage> messages) {
+        if (!K9.DEBUG) {
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/controller/MessagingController.java]: Despite the looks `K9.DEBUG` is not constant and tied to the "debug logging" setting. Please use `BuildConfig.DEBUG`.

File: k9mail/src/main/java/com/fsck/k9/fragment/MessageListFragment.java [hunk 2]
```diff
@@ -1435,6 +1441,7 @@ public void onCreateContextMenu(ContextMenu menu, View v, ContextMenuInfo menuIn
         }
 
         getActivity().getMenuInflater().inflate(R.menu.message_list_item_context, menu);
+        menu.findItem(R.id.debug_delete_locally).setVisible(K9.DEBUG);
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/fragment/MessageListFragment.java]: Same here: should be `BuildConfig.DEBUG`

File: k9mail/src/main/java/com/fsck/k9/mailstore/LocalMessage.java [hunk 3]
```diff
@@ -315,6 +316,39 @@ public Void doDbWork(final SQLiteDatabase db) throws WrappedException, Unavailab
         localStore.notifyChange();
     }
 
+    public void debugClearLocalData() throws MessagingException {
+        if (!K9.DEBUG) {
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/mailstore/LocalMessage.java]: Sorry, I missed this the last time. It should be `BuildConfig.DEBUG`, too.

Comment (Valodim) [inline k9mail/src/main/java/com/fsck/k9/mailstore/LocalMessage.java]: heh, if you had only put one comment I would probably have looked for others myself :P
